src/contains.js

```diff
@@ -0,0 +1,94 @@
+import {default as polygonContains} from "./polygonContains";
+import {default as distance} from "./distance";
+import {epsilon,radians} from "./math";
+
+var contains = {
+  Sphere: function() {
+    return true;
+  },
+  Point: function(feature, point) {
+    return distance(feature.coordinates, point) === 0;
+  },
+  MultiPoint: function(feature, point) {
+    var coordinates = feature.coordinates, i = -1, n = coordinates.length, f;
+    while (++i < n) {
+      f = { type: "Point", coordinates: coordinates[i] };
+      if (contains[f.type](f, point)) {
+        return true;
+      }
+    }
+    return false;
+  },
+  LineString: function(feature, point) {
+    var ab = distance(feature.coordinates[0], feature.coordinates[1]),
+      ao = distance(feature.coordinates[0], point),
+      ob = distance(point, feature.coordinates[1]);
+    return ao + ob <= ab + epsilon;
+  },
+  MultiLineString: function(feature, point) {
+    var coordinates = feature.coordinates, i = -1, n = coordinates.length, f;
+    while (++i < n) {
+      f = { type: "LineString", coordinates: coordinates[i] };
+      if (contains[f.type](f, point)) {
+        return true;
+      }
+    }
+    return false;
+  },
+  Polygon: function(feature, point) {
+    return polygonContains(feature.coordinates.map(ringRadians), pointRadians(point))
+      ? true
+      : false;
+  },
+  MultiPolygon: function(feature, point) {
+    var coordinates = feature.coordinates, i = -1, n = coordinates.length, f;
+    point = pointRadians(point);
+    while (++i < n) {
+      f = coordinates[i].map(ringRadians);
+      if (polygonContains(f, point)) {
+        return true;
+      }
+    }
+    return false;
+  },
+  GeometryCollection: function(feature, point) {
+    var geometries = feature.geometries, i = -1, n = geometries.length;
+    while (++i < n) {
+      var f = geometries[i];
+      if (contains[f.type](f, point)) {
+        return true;
+      }
+    }
+    return false;
+  },
+  Feature: function(feature, point) {
+    var geometry = feature.geometry;
+    return contains[geometry.type](geometry, point)
+      ? true
+      : false;
```

Comment:
This ternary expression shouldn’t be necessary since `contains[geometry.type]` should only return true or false.